mlir/lib/Conversion/AffineToStandard/AffineToStandard.cpp

```diff
@@ -334,7 +334,19 @@ class AffineYieldOpLowering : public OpRewritePattern<AffineYieldOp> {
 
   LogicalResult matchAndRewrite(AffineYieldOp op,
                                 PatternRewriter &rewriter) const override {
-    rewriter.replaceOpWithNewOp<scf::YieldOp>(op);
+    if (isa<scf::ParallelOp>(op.getParentOp())) {
+      // This could happen while lowering affine.parallel to scf.parallel.
+      // scf.parallel in all cases expect scf.yield to have no operands.
+      rewriter.replaceOpWithNewOp<scf::YieldOp>(op);
+      return success();
+    }
+    SmallVector<Value, 4> yieldOperands(op.operands().begin(),
+                                        op.operands().end());
+    if (yieldOperands.size() > 0) {
```

Comment:
Use `!empty()` instead of `size() > 0` - the former is more efficient and idiomatic.

Reply:
Done.